portal/src/main/webapp/js/src/oncoprint/setup.js

```diff
@@ -1338,12 +1338,21 @@ window.CreateCBioPortalOncoprintWithToolbar = function (ctr_selector, toolbar_se
 		    }
 		}));
 	    },
-	    'addGenesetTracks': function (genetic_profile_id, geneset_ids) {
+	    'addGenesetTracks': function (genetic_profile_id, geneset_ids, geneset_link_map) {
 		oncoprint.suppressRendering();
 		var track_ids = [];
-		var i, track_geneset_id, track_params, new_track_id;
+		var i, track_geneset_id, track_params, new_track_id, track_label_obj;
 		for (i = 0; i < geneset_ids.length; i++) {
 		    track_geneset_id = geneset_ids[i];
+		    track_label_obj = new String(track_geneset_id);
+		    track_label_obj.html_content = (
+			    // encode the string as the textual contents of an
+			    // HTML element
+			    $('<div>').text(track_label_obj).html()
+			    // Include zero-width spaces to allow line breaks
+			    // after punctuation in (typically long) gs names
+			    .replace(/_/g, '_&#8203;')
```

Comment:
Yes. The reason I am passing around HTML code is to insert zero-width spaces into the gene set ID (which can occasionally be a whole phrase in MSigDB's naming), indicating points at which the browser could split the ID over multiple lines instead of overflowing.